src/Language/PureScript/TypeChecker/Entailment.hs

```diff
@@ -316,6 +347,7 @@ entails SolverOptions{..} constraint context hints =
             -- Make a dictionary from subgoal dictionaries by applying the correct function
             mkDictionary :: Evidence -> Maybe [Expr] -> m Expr
             mkDictionary (NamedInstance n) args = return $ foldl App (Var n) (fold args)
+            mkDictionary RowUnionInstance _ = return $ Literal (ObjectLiteral [])
```

Comment:
We need a placeholder for the subgoal here. We can't just ignore the arguments list, since it might not be empty. If it's non-empty, the new dictionary needs to appear here because we might have deferred the constraint.